libs/s25main/world/GameWorldView.cpp

```diff
@@ -347,12 +347,18 @@ void GameWorldView::DrawNameProductivityOverlay(const TerrainRenderer& terrainRe
             if(!no)
                 continue;
 
+            Position curPos = GetWorld().GetNodePos(pt) - offset + curOffset;
+            curPos.y -= 22;
+
             // Is object not belonging to local player?
             if(no->GetPlayer() != gwv.GetPlayerId())
+            {
+                auto* attackAidImage = LOADER.GetImageN("map_new", 20000);
+                if(GetWorld().GetGGS().getSelection(AddonId::MILITARY_AID) == 2)
+                    if(gwv.GetNumSoldiersForAttack(pt) > 0) // soldiers available for attack?
+                        attackAidImage->DrawFull(curPos - DrawPoint(0, attackAidImage->getHeight()));
```

Comment:
Braces for multi-line code in branches:
```suggestion
                if(GetWorld().GetGGS().getSelection(AddonId::MILITARY_AID) == 2)
                {
                    if(gwv.GetNumSoldiersForAttack(pt) > 0) // soldiers available for attack?
                        attackAidImage->DrawFull(curPos - DrawPoint(0, attackAidImage->getHeight()));
                }
```